assets/component-image-with-text.css

```diff
@@ -299,6 +297,14 @@
   z-index: 2;
 }
 
+.image-with-text:not(.image-with-text--overlap) .color-background-1 {
+  background: transparent;
+}
+
+.image-with-text .gradient:not(.color-background-1) {
+  background-attachment: local;
+}
```

Comment:
This first style is to preserve some of the background-1 cases that are possible in image with text, like these..
<img src="https://screenshot.click/18-27-8pd6c-eb5q7.png" width="200" />

The second rule basically removes the fixed global gradient look for all non-background-1 cases so image with text displays the same with and without animations enabled https://screenshot.click/18-32-mrcd0-5ujji.mp4

Reply:
Can you remind me of the downside to making the background transparent elsewhere (like on Product cards) too? It seems like it would minimize the overall changes for merchants today. If I recall, we had some concerns about shadows overlapping text, but if merchants keep the shadows reasonable I think we'll be okay. 🤔 

<details>
<summary>Example</summary>

Before|After
---|---
<img width="1121" alt="Screenshot 2023-05-23 at 2 35 14 PM" src="https://github.com/Shopify/dawn/assets/1202812/c7786ab6-b72b-4372-9e37-d3d9eb9d7c94">|<img width="1122" alt="Screenshot 2023-05-23 at 2 36 03 PM" src="https://github.com/Shopify/dawn/assets/1202812/982ec318-0105-47a2-8c06-0e354108eb76">

</details>

Reply:
For cards I didn't think there were too many concerns other than potential shadow stuff. Though I thought we were interested in "correcting" the way gradients were applied to card style cards anyway so that the gradient was applied in a more expected way. I can be convinced of either. @ludoboludo any thoughts?